src/tracks/playabletrack/wavetrack/ui/WaveClipTrimHandle.cpp

```diff
@@ -64,48 +71,105 @@ class WaveClipTrimHandle::AdjustBorder final : public WaveClipTrimHandle::ClipTr
       }
    }
 
+   //Search for a good snap points among all tracks, including
+   //the one to which adjusted clip belongs to, but not counting
+   //the borders of adjusted clip
+   static SnapPointArray FindSnapPoints(
+      const WaveTrack* currentTrack, 
+      WaveClip* adjustedClip,
+      const std::pair<double, double> range)
+   {
+      SnapPointArray result;
+
+      auto addSnapPoint = [&](double t, const Track* track)
+      {
+         if(t > range.second || t < range.first)
+            return;
+
+         for(const auto& snapPoint : result)
+            if(snapPoint.t == t)
+               return;
+         result.emplace_back(t, track);
+      };
+
+      if(const auto trackList = currentTrack->GetOwner())
+      {
+         for(const auto track : trackList->Any())
+         {
+            if(track == currentTrack ||
+               track->GetLinkType() == Track::LinkType::Aligned && *trackList->FindLeader(currentTrack) == track ||
+               currentTrack->GetLinkType() == Track::LinkType::Aligned && *trackList->FindLeader(track) == currentTrack)
```

Comment:
Clang gives me a warning about '&&' within '||'.  Add parentheses to make the intent unmistakable.

Reply:
Funny, the tool I use to analyze a code suggests quite opposite: it says that parenthesizes are redundant :) But I usually tend to put them too. Thank you!